internal/resolution/entitysources/catalogdsource.go

```diff
@@ -0,0 +1,159 @@
+package entitysources
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+	"github.com/operator-framework/operator-registry/alpha/property"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	catalogd "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+)
+
+// catalogdEntitySource is a source for(/collection of) deppy defined input.Entity, built from content
+// made accessible on-cluster by https://github.com/operator-framework/catalogd.
+// It is an implementation of deppy defined input.EntitySource
+type catalogdEntitySource struct {
+	client client.Client
+}
+
+// type catalogdEntitySource map[deppy.Identifier]input.Entity
+
+func NewCatalogdEntitySource(ctx context.Context, client client.Client) catalogdEntitySource {
+
+	return catalogdEntitySource{client: client}
+}
+
+func (es catalogdEntitySource) Get(ctx context.Context, id deppy.Identifier) *input.Entity {
```

Comment:
Similarly, I think we want a pointer receiver, or these methods will run with a value-copy of `client`

Reply:
I'm going with a pointer receiver anyway since the struct will likely grow to include other stuff (eg a cache)